Братышев Тимофей: дз 1

TaimonFix opened the pull request.

TaimonFix committed:
Создание CRUD-операций

TaimonFix committed:
fix: проект запускается

TaimonFix committed:
Перенес БД на PostgreSQL

TaimonFix committed:
Добавил Docker контейнеризацию

TaimonFix committed:
Добавил аспект LoggingAspect:
- Реализовал @Before
- Добавил аннотацию @CustomLogging

TaimonFix committed:
Работа над реализацией @AfterThrowing

TaimonFix committed:
- добавил замер выполнения методов
- исправил ошибки

Alexandr056 commented on src/main/java/ru/online/t1_academy/projectT1/aspect/TrackingAspect.java:
```diff
@@ -0,0 +1,30 @@
+        } catch (Throwable e) {
+            throw new NullTaskException(e.getMessage());
+        }
+        long finish = System.currentTimeMillis();
```
зачем замерять время выполнения при исключении? сюда дойдем после блока catch…

Alexandr056 commented on src/main/java/ru/online/t1_academy/projectT1/controller/TaskController.java:
```diff
@@ -0,0 +1,76 @@
+@RequestMapping("/tasks")
+public class TaskController {
+    @Autowired
+    private TaskService service;
```
некорректный способ инжекта

Alexandr056 commented on src/main/java/ru/online/t1_academy/projectT1/controller/TaskController.java:
```diff
@@ -0,0 +1,76 @@
+
+@RestController
+@RequestMapping("/tasks")
+public class TaskController {
```
ResponseEntity в контроллерах лучше не использовать, более верный подход это ошибки в Service, а их обработка возможна в ExceptionHandler и ControllerAdvice

Alexandr056 commented on src/main/java/ru/online/t1_academy/projectT1/dto/Task.java:
```diff
@@ -0,0 +1,21 @@
+import jakarta.persistence.Id;
+import lombok.*;
+
+@Data
```
весьма не рекомендовано над Entity

Alexandr056 commented on src/main/java/ru/online/t1_academy/projectT1/dto/Task.java:
```diff
@@ -0,0 +1,21 @@
+@Entity
+@AllArgsConstructor
+@NoArgsConstructor
+public class Task {
```
добавьте Table аннотацию с указанием имени таблицы

Alexandr056 commented on src/main/java/ru/online/t1_academy/projectT1/service/TaskService.java:
```diff
@@ -0,0 +1,60 @@
+@Service
+public class TaskService {
+    @Autowired
+    private TaskRepository repository;
```
некорреткный способ инжекта

TaimonFix committed:
Рефакторинг кода:
- Исправил логику замера времени, теперь вся логика внутри блока try
- Добавил CustomExceptionHandler, чтобы убрать обработку ошибок в Controller
